src/PluginForm/ExpressCheckoutForm.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\commerce_paypal\PluginForm;
+
+use Drupal\commerce_payment\PluginForm\PaymentOffsiteForm as BasePaymentOffsiteForm;
+use Drupal\Core\Form\FormStateInterface;
+
+class ExpressCheckoutForm extends BasePaymentOffsiteForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    /** @var \Drupal\commerce_payment\Entity\PaymentInterface $payment */
+    $payment = $this->entity;
+    /** @var \Drupal\commerce_paypal\Plugin\Commerce\PaymentGateway\ExpressCheckoutInterface $payment_gateway_plugin */
+    $payment_gateway_plugin = $payment->getPaymentGateway()->getPlugin();
+
+    $extra = [
+      'return_url' => $form['#return_url'],
+      'cancel_url' => $form['#cancel_url'],
+      'capture' => $form['#capture'],
+    ];
+
+    $paypal_response = $payment_gateway_plugin->apiSetExpressCheckout($payment, $extra);
+
+    if (!empty($paypal_response['TOKEN'])) {
+      $order = $payment->getOrder();
+      $order->setData('paypal_express_checkout', [
+        'flow' => 'ec',
+        'token' => $paypal_response['TOKEN'],
+        'payerid' => FALSE,
+        'capture' => $extra['capture'],
+      ]);
+      $order->save();
+
+      // Redirect the buyer to PayPal URL.
+      if ($payment_gateway_plugin->getMode() == 'test') {
+        $redirect_url = 'https://www.sandbox.paypal.com/checkoutnow?token=' . $paypal_response['TOKEN'];
+      }
+      else {
+        $redirect_url = 'https://www.paypal.com/checkoutnow?token=' . $paypal_response['TOKEN'];
+      }
+    }
+    else {
+      $redirect_url = '';
```

Comment:
We need to cancel the build of the form here. This will cause `InvalidArgumentException: The URI '' is invalid. You must use a valid URI scheme. in Drupal\Core\Url::fromUri()` to be thrown, WSOD the checkout.

We should also try to get a PayPal response when validating the payment gateway's API information.